dango/types/src/lending.rs

```diff
@@ -0,0 +1,76 @@
+use {
+    grug::{Addr, Coins, Denom},
+    std::collections::BTreeSet,
+};
+
+/// The namespace that lending pool uses.
+pub const NAMESPACE: &str = "lending";
+
+#[grug::derive(Serde)]
+pub struct InstantiateMsg {
+    pub whitelisted_denoms: BTreeSet<Denom>,
+}
+
+#[grug::derive(Serde)]
+pub enum ExecuteMsg {
+    /// Whitelist a denom. Can only be invoked by the owner.
+    WhitelistDenom(Denom),
+
+    /// Delist a denom. Can only be invoked by the owner.
+    DelistDenom(Denom),
+
+    /// Deposit tokens into the lending pool.
+    Deposit {
+        /// The optional recipient of the minted LP tokens. If not set, the
+        /// sender's address will be used.
+        recipient: Option<Addr>,
+    },
+
+    /// Withdraw tokens from the lending pool by redeeming LP tokens. LP tokens
+    /// should be sent to the contract together with this message.
+    Withdraw {
+        /// The optional recipient of the withdrawn tokens. If not set, the
+        /// sender's address will be used.
+        recipient: Option<Addr>,
+    },
```

Comment:
I'm going to remove the optional `recipient`. Can't think of a use case anyone would use this. The less complexity the better

Reply:
Sure. I didn't have a direct use case in mind, but I find it to be a very nice pattern which enables composing contracts easily without going back and updating things.